### DISCUSSION TITLE
Is possible infer `TError` type from `queryFn` / `mutationFn` not using generic?

### DISCUSSION THREAD

**Discussion by zudydy** [TanStack/query#8403] [Q&A]
```ts
type CustomResponse = {
  // some field...
};

type CustomError = {
  // some field...
};

const fetchSomething = async () => {
  try {
    const { data } = await axios.get<CustomResponse>("/some-url");
    return data;
  } catch (error) {
    throw error as AxiosError<CustomError>;
  }
};

const { data, error } = useQuery({
  queryKey: ["some-key"],
  queryFn: fetchSomething,
})
```

In this situation, the type of `data` is inferred as `CustomResponse`, while the type of `error` is not inferred as `CustomError`.

I only want to specify the response type and error type for the API fetcher function part, and I don't want to enter any generics for the query call part.
I just want to make `data` and `error` types of inference just by passing the API fetcher function to `queryFn` or `mutationFn`.

Is there a way to do this? Or is the `TError` type designed to be impossible to infer with `queryFn` or `mutationFn`?

Thx :)

**Comment by TkDodo** [comment]
https://tanstack.com/query/v5/docs/framework/react/typescript#registering-a-global-error
